### ISSUE TITLE
Support Model Bound Attributes to add select/expand limits, etc

### ISSUE THREAD

**Issue by techniq** [voronov-maxim/OdataToEntity#29]
Currently using OData/WebApi, I use the `ODataConventionModelBuilder` to build the `EdmModel`, but this also gives me access to the entity types before building the `EdmModel`, which allows me to add restrictions (we keep our model constrained as much as possible, especially with regards to expanding through properties.

For example:
```c#
var builder = new ODataConventionModelBuilder(serviceProvider);

builder
    .EntitySet<Department>("Departments")
    .EntityType
    .Expand(1, x => x.Organization)
    .Expand(3, x => x.CategoryMembership)
    .IgnoreExcept(c => new
    {
        c.Id,
        c.Code,
        c.Name,
        c.OrganizationId,
        c.Organization,
        c.Roles,
        c.CategoryMembership
    });

builder
  .EntitySet<Employee>("Employees")
  .EntityType
  .Expand(2, x => x.Department)
  .Expand(1, x => x.Position)
  .IgnoreExcept(c => new
  {
      c.Id,
      c.FirstName,
      c.MiddleName,
      c.LastName,
      c.DepartmentId,
      c.Department,
      c.PositionId,
      c.Position
  });

// ...

return builder.GetEdmModel();
```

After some reading, I think this could be best accomplished in ODataToEntity by supporting the [Model Bound Attributes](http://odata.github.io/WebApi/#13-01-modelbound-attribute).  For example (pulled from referenced page):

```c#
[Expand("Orders", "Friend", "CountableOrders", MaxDepth = 10)]
[Expand("AutoExpandOrder", ExpandType = SelectExpandType.Automatic, MaxDepth = 8)]
[Page(MaxTop = 5, PageSize = 1)]
public class Customer
{
    public int Id { get; set; }
    public string Name { get; set; }
    [Expand(ExpandType = SelectExpandType.Disabled)]
    public Order Order { get; set; }
    public Order AutoExpandOrder { get; set; }
    public Address Address { get; set; }
    [Expand("Customers", MaxDepth = 2)]
    [Count(Disabled = true)]
    [Page(MaxTop = 2, PageSize = 1)]
    public List<Order> Orders { get; set; }
    public List<Order> CountableOrders { get; set; }
    public List<Order> NoExpandOrders { get; set; }
    public List<Address> Addresses { get; set; }
    [Expand(MaxDepth = 2)]
    public Customer Friend { get; set; }
}
```
which would match this Fluent API approach (just here for reference)
```c#
var builder = new ODataConventionModelBuilder();
builder.EntitySet<Customer>("Customers")
    .EntityType.Expand(10, "Orders", "Friend", "CountableOrders")
    .Expand(8, SelectExpandType.Automatic, "AutoExpandOrder")
    .Page(5, 2);
builder.EntityType<Customer>()
    .HasMany(p => p.Orders)
    .Expand(2, "Customers")
    .Page(2, 1)
    .Count(QueryOptionSetting.Disabled);
builder.EntityType<Customer>()
    .HasMany(p => p.CountableOrders)
    .Count();
builder.EntityType<Customer>()
    .HasOptional(p => p.Order)
    .Expand(SelectExpandType.Disabled);
```

**Comment by techniq** [comment]
I just checked and these attributes are part of [OData/WebApi](https://github.com/OData/WebApi/tree/d02bc61ea7b31ada1e54abbeebbecb3c5df0e3ac/src/Microsoft.AspNet.OData.Shared/Query) and not ODataLib like I had hoped, so to support this method, the attributes would need to be replicated in ODataToEntity (or support another approach, like fluent API access to the EdmModel builder).

In some cases, both attributes and a fluent API would be nice, as attributes can be nice, but you don't always have access to the models (this used to be the case for us as we had shared models across projects, each with different data access constraints).  We opted to replicate the models into each project (only adding the classes, and properties, needed for each project), but a fluent API would allow defining these separately.
